src/test/java/com/endava/intership/collections/StudentMapTest.java

```diff
@@ -0,0 +1,189 @@
+package com.endava.intership.collections;
+
+import com.endava.internship.collections.Student;
+import com.endava.internship.collections.StudentMap;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class StudentMapTest {
+
+    private static StudentMap<Student, Integer> testTreeMap;
+    private static ArrayList<Student> studentsList;
+
+    @BeforeAll
+    static void init() {
+        testTreeMap  = new StudentMap<>();
+        studentsList = new ArrayList<>();
+
+        studentsList.add(new Student("Neiculov Alexei", LocalDate.of(2002, 2, 18), "CEITI"));
+        studentsList.add(new Student("Terzi Kirill", LocalDate.of(2002, 12, 11), "CEITI"));
+        studentsList.add(new Student("Ponomar Dmitri", LocalDate.of(2003, 7, 8), "ASEM"));
+        studentsList.add(new Student("Rotaru Timon", LocalDate.of(2004, 5, 16), "UTM"));
+        studentsList.add(new Student("Ivanov Victor", LocalDate.of(1994, 4, 3), "UTM"));
+        studentsList.add(new Student("Boblic Stefan", LocalDate.of(2001, 5, 9), "ASEM"));
+        studentsList.add(new Student("Boblic Stefan", LocalDate.of(2000, 3, 6), "ASEM"));
+        studentsList.add(new Student("Karaman Anatolii", LocalDate.of(2002, 8, 7), "USM"));
+        studentsList.add(new Student("Simlih Ilia", LocalDate.of(2002, 8, 2), "CEITI"));
+
+        for (int i = 0; i < studentsList.size(); i++) {
+            testTreeMap.put(studentsList.get(i), i+1);
+        }
+    }
+
+    private static Stream<Arguments> provideStudentsForTreeMapTests() {
+        ArrayList<Arguments> arguments = new ArrayList<>();
+        studentsList.forEach(student -> arguments.add(Arguments.of(student, studentsList.indexOf(student)+1)));
+        return arguments.stream();
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideStudentsForTreeMapTests")
+    void putTest(Student student, int number) {
+        assertAll(
+                () -> assertThat(testTreeMap).containsKey(student),
+                () -> assertThat(testTreeMap.get(student)).isEqualTo(number),
+                () -> assertThat(testTreeMap).containsValue(number)
+        );
+    }
+
+    @Test
+    void putOneTest() {
+        Student expectedStudent = new Student("Neiculov Alexei", LocalDate.of(2002, 2, 18), "CEITI");
+        StudentMap<Student, Integer> testTreeMap = new StudentMap<>();
+        testTreeMap.put(expectedStudent, 50);
+
+        assertAll(
+                () -> assertThat(testTreeMap).hasSize(1),
+                () -> assertThat(testTreeMap).isNotEmpty(),
+                () -> assertThat(testTreeMap).containsKey(expectedStudent),
```

Comment:
you can make this assert more accurate: assertEquals(testTreeMap.get(0).getKey(value), expectedStudent);
but both ways are valid

Reply:
ok